experimenter/tests/integration/nimbus/test_cirrus_integration.py

```diff
@@ -158,3 +158,95 @@ def test_create_new_experiment_approve_remote_settings_cirrus(
     # returns the default value
     result_text_element = demo_app.wait_for_result_text(["wicked"])
     assert result_text_element.is_displayed()
+
+
+@pytest.mark.cirrus_enrollment
+def test_check_cirrus_targeting(
+    selenium,
+    experiment_url,
+    create_experiment,
+    kinto_client,
+    base_url,
+    experiment_name,
+    demo_app,
+):
+    # Launch an experiment with two branches
+    reference_branch_value = '{"enabled": true, "something": "Control branch"}'
+    treatment_branch_value = '{"enabled": true, "something": "Treatment branch"}'
+    create_experiment(
+        selenium,
+        reference_branch_value=reference_branch_value,
+        treatment_branch_value=treatment_branch_value,
+        languages=True,
+        countries=True,
+    ).launch_and_approve()
+
+    kinto_client.approve()
+
+    SummaryPage(selenium, experiment_url).open().wait_for_live_status()
+
+    home = HomePage(selenium, base_url).open()
+    assert True in [experiment_name in item.text for item in home.tables[0].experiments]
+
+    # Demo app frontend, by default, returns "Not Enrolled" message
+    navigate_to(selenium)
+    selenium.refresh()
+
+    result_text_element = demo_app.wait_for_result_text(["Not Enrolled"])
+    assert result_text_element.is_displayed()
+
+    # Pass client_id and context
+    demo_app.fill_and_send_form_data("test", '{"language":"en", "region":"CA"}')
+    demo_app.click_send_my_details()
+
+    # Determine the variation displayed and assert accordingly
+    # Check if either "Control branch" or "Treatment branch" is displayed
+    displayed_text = demo_app.wait_for_result_text(
+        ["Control branch", "Treatment branch"]
+    ).text
+    assert displayed_text in ["Control branch", "Treatment branch"]
+
+    # Refresh the page and try passing a new client
+    selenium.refresh()
+    result_text_element = demo_app.wait_for_result_text(["Not Enrolled"])
+    assert result_text_element.is_displayed()
+
+    demo_app.fill_and_send_form_data("example1", '{"language":"en", "region":"CA"}')
+    demo_app.click_send_my_details()
+
+    # Determine the variation displayed and assert accordingly
+    displayed_text = demo_app.wait_for_result_text(
+        ["Control branch", "Treatment branch"]
+    ).text
+    assert displayed_text in ["Control branch", "Treatment branch"]
+
+    # Unenroll
+    summary = SummaryPage(selenium, experiment_url).open()
+    summary.end_and_approve()
+    kinto_client.approve()
+    summary = SummaryPage(selenium, experiment_url).open()
+    summary.wait_for_complete_status()
```

Comment:
Oh, opening the summary page directly like this is one of the things that's caused intermittency in the past.  It might be fine here, but just watch out if you hit these points and intermittently get someting like unable to find end status button, etc that might be the cause?

Reply:
my backend support was missing, I hope that error should be gone now

Reply:
The ```open()``` function _should_ call the appropriate wait method.